src/ButtonStrip.js

```diff
@@ -14,9 +28,27 @@ import styles from './ButtonStrip/styles.js'
  */
 const ButtonStrip = ({ className, children, middle, end }) => (
     <div className={cx(className, { middle, end })}>
-        {children}
-
-        <style jsx>{styles}</style>
+        {Children.map(children, child =>
+            cloneElement(child, {
+                className: cx(child.props.className, buttonStyles.className, {
+                    start: !middle && !end,
+                    middle,
+                    end,
+                }),
+            })
+        )}
```

Comment:
Code complexity increased quite a bit here to switch to style-overrides via resolve. Since the risk of styles bleeding in via the generic `start`/`middle`/`end` classNames is still very real, I wonder if this is worth it.